File: services-turf/src/test/java/org/maplibre/turf/TurfClassificationTest.java
package org.maplibre.turf;

import org.maplibre.geojson.Feature;
import org.maplibre.geojson.Point;
import org.maplibre.geojson.FeatureCollection;

import org.junit.Assert;
import org.junit.Test;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;

public class TurfClassificationTest extends TestUtils {

  private static final String PT = "turf-classification/pt.json";
  private static final String PTS = "turf-classification/pts.json";

  @Test
  public void testLineDistanceWithGeometries() throws IOException, TurfException {
    Point pt = (Point) Feature.fromJson(loadJsonFixture(PT)).geometry();
    FeatureCollection pts = FeatureCollection.fromJson(loadJsonFixture(PTS));

    List<Point> pointList = new ArrayList<>();
    for (Feature feature : pts.features()) {
      pointList.add((Point) (feature.geometry()));
    }
    Point closestPt = TurfClassification.nearestPoint(pt, pointList);

    Assert.assertNotNull(closestPt);
    Assert.assertEquals(closestPt.type(), "Point");
    Assert.assertEquals(closestPt.longitude(), -75.33, DELTA);
    Assert.assertEquals(closestPt.latitude(), 39.44, DELTA);
  }
}

File: services-turf/src/test/java/com/mapbox/turf/TurfClassificationTest.kt
package com.mapbox.turf

import com.mapbox.geojson.Feature
import com.mapbox.geojson.FeatureCollection
import com.mapbox.geojson.Point
import com.mapbox.turf.TurfClassification.nearestPoint
import org.junit.Assert
import org.junit.Test
import java.io.IOException

class TurfClassificationTest : TestUtils() {
    @Test
    @Throws(IOException::class, TurfException::class)
    fun testLineDistanceWithGeometries() {
        val pt = Feature.fromJson(loadJsonFixture(PT)).geometry() as Point?
        val pts = FeatureCollection.fromJson(loadJsonFixture(PTS))
        val pointList: MutableList<Point> = ArrayList()
        for (feature in pts.features()!!) {
            pointList.add(feature.geometry() as Point)
        }
        val closestPt = nearestPoint(pt!!, pointList)
        Assert.assertNotNull(closestPt)
        Assert.assertEquals(closestPt.type(), "Point")
        Assert.assertEquals(closestPt.longitude(), -75.33, DELTA)
        Assert.assertEquals(closestPt.latitude(), 39.44, DELTA)
    }

    companion object {
        private const val PT = "turf-classification/pt.json"
        private const val PTS = "turf-classification/pts.json"
    }
}